src/commands/lambda/functions/instrument.ts

```diff
@@ -134,17 +146,49 @@ export const calculateUpdateRequest = async (
   }
   let needsUpdate = false
 
-  // Update Handler
-  const expectedHandler = HANDLER_LOCATION[runtime]
-  if (config.Handler !== expectedHandler) {
-    needsUpdate = true
-    updateRequest.Handler = HANDLER_LOCATION[runtime]
+  // If a layer version is set for Java or a custom runtime.
+  // We need to inform the customer to only set the extension argument.
+  if (RUNTIME_LOOKUP[runtime] === RuntimeType.JAVA || RUNTIME_LOOKUP[runtime] === RuntimeType.CUSTOM) {
+    if (settings.layerVersion !== undefined) {
+      throw new Error(
+        `Only the extension version should be set for the ${runtime} runtime. Please remove the layer version from the instrument command and only use the extension version.`
```

Comment:
I think customers could get confused when we tell them to remove the "layer version". Technically the extension is a layer too. Maybe we should be clear that `-v` is a runtime-specific layer version, and `-e` is the extension layer version. It's the runtime-specific layer that we can't add in these runtimes using Datadog CI.